internal/server/data/provideruser_test.go

```diff
@@ -254,3 +255,108 @@ func TestDeleteProviderUser(t *testing.T) {
 		assert.NilError(t, err)
 	})
 }
+
+func TestListProviderUsers(t *testing.T) {
+	type testCase struct {
+		name  string
+		setup func(t *testing.T, tx *Transaction) (providerID uid.ID, p *SCIMParameters, expected []models.ProviderUser)
+	}
+
+	testCases := []testCase{
+		{
+			name: "list all provider users",
+			setup: func(t *testing.T, tx *Transaction) (providerID uid.ID, p *SCIMParameters, expected []models.ProviderUser) {
+				provider := &models.Provider{
+					Name: "mockta",
+					Kind: models.ProviderKindOkta,
+				}
+
+				err := CreateProvider(tx, provider)
+				assert.NilError(t, err)
+
+				pu := createTestProviderUser(t, tx, provider, "david@example.com")
+				return provider.ID, nil, []models.ProviderUser{pu}
+			},
+		},
+		{
+			name: "list all provider users invalid provider ID",
+			setup: func(t *testing.T, tx *Transaction) (providerID uid.ID, p *SCIMParameters, expected []models.ProviderUser) {
+				provider := &models.Provider{
+					Name: "mockta",
+					Kind: models.ProviderKindOkta,
+				}
+
+				err := CreateProvider(tx, provider)
+				assert.NilError(t, err)
+
+				_ = createTestProviderUser(t, tx, provider, "david@example.com")
+				return 1234, nil, nil
+			},
+		},
+		{
+			name: "limit less than total",
+			setup: func(t *testing.T, tx *Transaction) (providerID uid.ID, p *SCIMParameters, expected []models.ProviderUser) {
+				provider := &models.Provider{
+					Name: "mockta",
+					Kind: models.ProviderKindOkta,
+				}
+
+				err := CreateProvider(tx, provider)
+				assert.NilError(t, err)
+
+				pu := createTestProviderUser(t, tx, provider, "david@example.com")
+				_ = createTestProviderUser(t, tx, provider, "lucy@example.com")
+				return provider.ID, &SCIMParameters{Count: 1}, []models.ProviderUser{pu}
+			},
+		},
+		{
+			name: "offset from start",
+			setup: func(t *testing.T, tx *Transaction) (providerID uid.ID, p *SCIMParameters, expected []models.ProviderUser) {
+				provider := &models.Provider{
+					Name: "mockta",
+					Kind: models.ProviderKindOkta,
+				}
+
+				err := CreateProvider(tx, provider)
+				assert.NilError(t, err)
+
+				_ = createTestProviderUser(t, tx, provider, "david@example.com")
+				pu := createTestProviderUser(t, tx, provider, "lucy@example.com")
+				return provider.ID, &SCIMParameters{StartIndex: 1}, []models.ProviderUser{pu}
+			},
+		},
+	}
+
+	runDBTests(t, func(t *testing.T, db *DB) {
+		org := &models.Organization{Name: "something", Domain: "example.com"}
+		assert.NilError(t, CreateOrganization(db, org))
+
+		for _, tc := range testCases {
+			t.Run(tc.name, func(t *testing.T) {
+				tx := txnForTestCase(t, db, org.ID)
+
+				providerID, p, expected := tc.setup(t, tx)
+
+				result, err := ListProviderUsers(tx, providerID, p)
+
+				assert.NilError(t, err)
+				assert.DeepEqual(t, result, expected, cmpTimeWithDBPrecision)
```

Comment:
It'd be great to include an assert for the `TotalCount` as well, which gets set on `p` I think.

Reply:
Added some assertions here